### PR TITLE
cmd/kubectl-gadget/top: Use `BaseParser` to uniform and simplify gadgets

### PR DESCRIPTION
**Pull request by blanquicet**
# Use `BaseParser` to uniform and simplify top category gadgets

This PR modifies the top category gadgets to use `BaseParser` so that they don't have to implement the logic to print the header and transform stats to columns.

This PR doesn't modify the logic or current behaviour of the gadgets. This PR aims to uniform this category with the rest of the categories so we can manage the two mentioned logics from a single place for all the gadgets. For instance, combine the `OutputModeColumns` and `OutputModeCustomColumns`, and printing the columns that can be used with the `custom-columns` flag.

## How to use

These changes don't have any impact on the user.

### CODE REVIEW

**Review by mauriciovasquezbernal — COMMENTED** [review]
I like it. I'm wondering if you are planning to create a `TopParser` to reduce the duplicated logic in those gadgets?

File: cmd/kubectl-gadget/top/block-io.go
Comment (blanquicet) [reply cmd/kubectl-gadget/top/block-io.go:222]: I was wondering the same when I modified this code. We were already applying this feature only for non-JSON output mode.

I can enable it for the JSON output mode as well but I think we should improve the description of this flag because "Maximum rows to print" doesn't make sense for the JSON output mode. WDYT?

Comment (mauriciovasquezbernal) [reply cmd/kubectl-gadget/top/block-io.go:222]: I think the confusion is because "row" = "item" in the columns output mode. I don't have ideas about to make this description better, without renaming it to something else. I think it's just fine to reduce the number of generated items in the json if you're passing `--maxRows` and `-o json`.

File: cmd/kubectl-gadget/top/block-io.go
Comment (flyth) [inline cmd/kubectl-gadget/top/block-io.go:85]: Those information would be really good to have available more generically (through an interface for example) - but that's probably out of scope here.

**Review by flyth — COMMENTED** [review]
Tested - works and looks good :)

File: cmd/kubectl-gadget/top/block-io.go
Comment (blanquicet) [reply cmd/kubectl-gadget/top/block-io.go:85]: Absolutely agree! Let's explore some ideas for a future PR.

File: cmd/kubectl-gadget/top/block-io.go
Comment (blanquicet) [reply cmd/kubectl-gadget/top/block-io.go:37]: Make sense in the theory, but I don't see how to do that as `CommonTopFlags.SortBy` is used to store the flag [here](https://github.com/kinvolk/inspektor-gadget/blob/jose/use-parser-top-cat/cmd/kubectl-gadget/top/top.go#L54). Do you have any suggestion?

File: cmd/kubectl-gadget/top/block-io.go
Comment (blanquicet) [reply cmd/kubectl-gadget/top/block-io.go:127]: Oops

File: cmd/kubectl-gadget/top/block-io.go
Comment (flyth) [reply cmd/kubectl-gadget/top/block-io.go:37]: Hmm. You could use `command.Flags().String(...)` in `addCommonTopFlags` and then in `PreRunE` use `cmd.Flags().GetString("sort")` - that helps with the ambiguity but doesn't really make it nicer.

Comment (blanquicet) [reply cmd/kubectl-gadget/top/block-io.go:37]: Mmmm it would prevent us from defining `SortBy` and also `ParsedSortBy` but it am not sure I like treating some flags in one way (`StringVarP`) and some others in a different way (`String`). IMO, it is not worth it.

Comment (flyth) [reply cmd/kubectl-gadget/top/block-io.go:37]: Agree. Just for the record (and because it might become handy or we change it later on), there would be yet another way: one could use `command.Flags().Var(...)` and then a variable of type `types.SortBy` directly, but would need to implement the `Value` interface for types.SortBy:

```
type Value interface {
	String() string
	Set(string) error
	Type() string
}
```

Comment (blanquicet) [reply cmd/kubectl-gadget/top/block-io.go:37]: TIL. Thanks for sharing it.

**Comment by blanquicet** [comment]
> I like it. I'm wondering if you are planning to create a `TopParser` to reduce the duplicated logic in those gadgets?

@mauriciovasquezbernal I would like to, but as we agreed [here](https://kubernetes.slack.com/archives/CSYL75LF6/p1655909355296489?thread_ts=1655287566.910119&cid=CSYL75LF6), I was waiting for the outcomes of the discussion about the possible change of the GlobalTracer/API, which may change the implementation of the CLI for those gadgets [#673](https://github.com/kinvolk/inspektor-gadget/issues/673). Given that such a task is taking more time than expected, do we want to unblock the refactoring of this category?

**Comment by blanquicet** [comment]
I rebased the PR to solve conflicts created after merging https://github.com/kinvolk/inspektor-gadget/pull/825.

File: cmd/kubectl-gadget/top/top.go
Comment (mauriciovasquezbernal) [inline cmd/kubectl-gadget/top/top.go:54]: I know it wasn't introduced in this PR, but `possible values are` seems too verbose. What about?

```suggestion
	command.Flags().StringVarP(&commonTopFlags.SortBy, "sort", "", sortBySlice[0], fmt.Sprintf("Sort column (%s)",  strings.Join(sortBySlice, ", ")))
```

File: cmd/kubectl-gadget/top/top.go
Comment (mauriciovasquezbernal) [inline cmd/kubectl-gadget/top/top.go:41]: nit: Could you leverage the opportunity to order this alphabetically?

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM.

**Pull request merged**
